config/buffer_period_test.go

```diff
@@ -156,33 +156,115 @@ func TestBufferPeriodConfig_Finalize(t *testing.T) {
 	cases := []struct {
 		name string
 		i    *BufferPeriodConfig
+		p    *BufferPeriodConfig
 		r    *BufferPeriodConfig
 	}{
 		{
-			"empty",
+			"empty input, parent enabled",
+			&BufferPeriodConfig{},
+			DefaultBufferPeriodConfig(),
+			&BufferPeriodConfig{
+				Enabled: Bool(true),
+				Min:     TimeDuration(5 * time.Second),
+				Max:     TimeDuration(20 * time.Second),
+			},
+		},
+		{
+			"empty input, parent disabled",
 			&BufferPeriodConfig{},
 			&BufferPeriodConfig{
 				Enabled: Bool(false),
+				Min:     TimeDuration(0 * time.Second),
+				Max:     TimeDuration(0 * time.Second),
+			},
+			&BufferPeriodConfig{
+				Enabled: Bool(false),
+				Min:     TimeDuration(0 * time.Second),
+				Max:     TimeDuration(0 * time.Second),
+			},
+		},
+		{
+			"disabled input",
+			&BufferPeriodConfig{
+				Enabled: Bool(false),
+			},
+			DefaultBufferPeriodConfig(),
+			&BufferPeriodConfig{
+				Enabled: Bool(false),
+				Min:     TimeDuration(0 * time.Second),
+				Max:     TimeDuration(0 * time.Second),
+			},
+		},
+		{
+			"enabled input, enabled parent",
+			&BufferPeriodConfig{
+				Enabled: Bool(true),
+			},
+			DefaultBufferPeriodConfig(),
+			&BufferPeriodConfig{
+				Enabled: Bool(true),
+				Min:     TimeDuration(5 * time.Second),
+				Max:     TimeDuration(20 * time.Second)},
+		},
+		{
+			"enabled input, disabled parent",
+			&BufferPeriodConfig{
+				Enabled: Bool(true),
+			},
+			&BufferPeriodConfig{
+				Enabled: Bool(false),
+				Min:     TimeDuration(0 * time.Second),
+				Max:     TimeDuration(0 * time.Second),
+			},
+			&BufferPeriodConfig{
+				Enabled: Bool(true),
 				Min:     TimeDuration(5 * time.Second),
 				Max:     TimeDuration(20 * time.Second),
 			},
 		},
 		{
-			"with_min",
+			"only min input",
 			&BufferPeriodConfig{
 				Min: TimeDuration(10 * time.Second),
 			},
+			DefaultBufferPeriodConfig(),
 			&BufferPeriodConfig{
 				Enabled: Bool(true),
 				Min:     TimeDuration(10 * time.Second),
 				Max:     TimeDuration(40 * time.Second),
 			},
 		},
+		{
+			"only max input",
+			&BufferPeriodConfig{
+				Max: TimeDuration(50 * time.Second),
+			},
+			DefaultBufferPeriodConfig(),
+			&BufferPeriodConfig{
+				Enabled: Bool(true),
+				Min:     TimeDuration(5 * time.Second),
+				Max:     TimeDuration(50 * time.Second),
+			},
+		},
+		{
+			"input and parent configured",
```

Comment:
Thanks for the added edge cases! While we're focused on this, could you add one more case where the input is enabled with values, and the global is disabled with values? It would be useful to test that not only the `Enabled` is overriding correctly, but also the min/max.

```go
"input enabled and parent disabled with values"
i: {
  Enabled: true
  Min: 20s
  Max: 50s
},
p: {
  Enabled: false
  Min: 1m
  Max: 5m
}
r: {
  // same as i
}
```

Reply:
Yes, added! Thank you